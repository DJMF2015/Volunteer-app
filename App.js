import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import Theme from './styles/Theme'
import ProjectList from './screens/ProjectList';

export default function App() {
  return (
    <Theme>
      <ProjectList />
    </Theme>

  );
}


