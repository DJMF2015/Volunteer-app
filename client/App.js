import React, {useState, useEffect} from 'react';
import {Text, SafeAreaView} from 'react-native';
import Theme from './styles/Theme';
import ProjectList from './screens/ProjectList';


export default function App() {
  
  const [data, setData] = useState(null);

  // To fecth from Firebase API:
  // const url = 'https://sva-api-default-rtdb.europe-west1.firebasedatabase.app/.json'
  // setData(info.projects)
  //
  // To fetch from Node server:
  const url = 'http://localhost:5000/projects'
  // 'setData(info)'


  useEffect( () => {
  fetch(url)
  .then(results => results.json())
  .then(info => setData(info))}, []);

  if (data) {
  return (
    <Theme>
      <ProjectList data = {data}/>
    </Theme>
  )
  }
  else {
  return (
    <SafeAreaView>
      <Text>Loading...</Text>
    </SafeAreaView>
  )
  }
}
