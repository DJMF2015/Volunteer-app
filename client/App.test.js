import React from 'react';
//import renderer from 'react-test-renderer';
import { render } from '@testing-library/react-native';

import App from './App';

// describe('<App />', () => {
//   it('has 5 children', () => {
//     const tree = renderer.create(<App />).toJSON();
//     expect(tree.children.length).toBe(5);
//   });
// });

describe('App component', () => {
  beforeEach(() => {
    fetch.resetMocks()
  })
  it('shout render default opportunity list', () => {
    const { getByText, toJSON } = render(<App />);
    expect(getByText('All projects')).toBeDefined();
    expect(getByText('Saved Projects')).toBeDefined();
    expect(getByText('My Projects')).toBeDefined();
    expect(getByText('Project Title')).toBeDefined();
    expect(getByText('STA internal')).toBeDefined();
    expect(getByText('Lead Test Analyst')).toBeDefined();
    expect(getByText(/The Lead Tester is a/)).toBeDefined();
    // TODO test interaction
    expect(toJSON()).toMatchSnapshot();
  });
});
