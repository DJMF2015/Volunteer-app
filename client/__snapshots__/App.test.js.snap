// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render default opportunity list 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#E5E5E5",
        "color": "#3c3c3b",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "color": "#3c3c3b",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "height": 81,
            "marginLeft": 16,
            "marginTop": 10,
            "width": 81,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "marginRight": 16,
            "marginTop": 10,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "marginRight": 54,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 38,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 23,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 23,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 18,
            },
          ]
        }
      >
        All projects
      </Text>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 18,
            },
          ]
        }
      >
        Saved Projects
      </Text>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 18,
            },
          ]
        }
      >
        My Projects
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderColor": "#3c3c3b",
          "borderStyle": "solid",
          "borderWidth": 1,
          "marginBottom": 10,
          "marginLeft": 75,
          "marginRight": 75,
          "marginTop": 0,
        },
      ]
    }
  />
  <RCTScrollView
    style={
      Array [
        Object {
          "height": "60%",
        },
      ]
    }
  >
    <View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#3c3c3b",
            "borderStyle": "solid",
            "borderWidth": 2,
            "marginBottom": 0,
            "marginLeft": 21,
            "marginRight": 41,
            "marginTop": 21,
            "opacity": 1,
            "paddingBottom": 11,
            "paddingLeft": 27,
            "paddingRight": 27,
            "paddingTop": 17,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "700",
                },
              ]
            }
          >
            STA-volunteer App
          </Text>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "fontSize": 16,
                "fontWeight": "400",
              },
            ]
          }
        >
          STA - internal
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontSize": 16,
                "fontWeight": "600",
                "marginTop": 9,
              },
            ]
          }
        >
          Lead developer
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontSize": 10,
                "fontWeight": "400",
                "marginTop": 4,
              },
            ]
          }
        >
          The lead tester is a coordination and management role, so an understanding of and experience in a number of testing disciplines, rather than depth in any specific one.
        </Text>
        <View
          style={
            Array [
              Object {
                "color": "#3c3c3b",
                "display": "flex",
                "flexDirection": "row",
                "marginTop": 17,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 16,
                },
                undefined,
                Object {
                  "fontFamily": "entypo",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "borderColor": "#3c3c3b",
                  "borderRadius": 4,
                  "borderStyle": "solid",
                  "borderWidth": 2,
                  "marginLeft": 8.5,
                  "paddingBottom": 0,
                  "paddingLeft": 1,
                  "paddingRight": 1,
                  "paddingTop": 0,
                },
              ]
            }
          >
            React Native
          </Text>
          <Text
            style={
              Array [
                Object {
                  "borderColor": "#3c3c3b",
                  "borderRadius": 4,
                  "borderStyle": "solid",
                  "borderWidth": 2,
                  "marginLeft": 8.5,
                  "paddingBottom": 0,
                  "paddingLeft": 1,
                  "paddingRight": 1,
                  "paddingTop": 0,
                },
              ]
            }
          >
            Node.js
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginTop": 7,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 16,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome5Free-Regular",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {
                  "fontWeight": "400",
                },
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "marginLeft": 13,
                },
              ]
            }
          >
            5-10 hours per week
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginTop": 7,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 16,
                },
                undefined,
                Object {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "marginLeft": 13,
                },
              ]
            }
          >
            One person required
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginTop": 7,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 16,
                },
                undefined,
                Object {
                  "fontFamily": "feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Array [
                Object {
                  "marginLeft": 13,
                },
              ]
            }
          >
            Suitable for buddying
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
